### ISSUE TITLE
RAM usage of egui

### ISSUE THREAD

**Issue by ivxvm** [emilk/egui#3689]
It's quite big. Not as big as Electron or similar WebView tech, but still too big for what it is.
My fairly simple app with few buttons and few labels eats 30mb RAM with strip enabled.
Similar app in fltk-rs consumes 1mb RAM!
Aren't immediate mode UIs expected to eat less RAM than retained mode UIs?
Are there any configuration adjustments or tricks to improve RAM usage of egui apps?
Compiling to 32bit architecture doesn't change anything.

**Comment by emilk** [comment]
I used `re_memory` to track the memory usage of `examples/hello_world_simple` on latest `main` with:

```rs
use re_memory::AccountingAllocator;

#[global_allocator]
static GLOBAL: AccountingAllocator<std::alloc::System> =
    AccountingAllocator::new(std::alloc::System);
```

And then showing it with:

```rs
let mem_use = re_memory::MemoryUse::capture();
if let Some(counted) = mem_use.counted {
    ui.label(format!(
        "Counted memory: {}",
        re_format::format_bytes(counted as _)
    ));
if let Some(resident) = mem_use.resident {
    ui.label(format!(
        "Resident memory: {}",
        re_format::format_bytes(resident as _)
    ));
}
```

`re_memory::AccountingAllocator` keeps track of _every_ dynamic allocation made by the application.

The results on my MacBook:

Counted memory: 3.1 MiB
Resident memory 35.9 MiB

The "counted" memory are all RAM allocations made by Rust.
Resident is the amount of memory used by the application, as reported by the OS.

In other words: egui itself is using 3.1 MiB of memory (or less). About 2MiB of that is the font texture. Another 1MiB is the default egui icon (which I admit seems a bit excessive).

I don't know where the rest of the overhead comes from - I'm guessing it is a combination of OS overhead, system allocator overhead, GPU memory allocated, etc. I don't know - but it isn't the `egui` crate.

There is also the `data` block of the application. egui embed around 1MiB of `.ttf` files, so maybe that's counted too.

**Comment by ivxvm** [comment]
Thanks for detailed answer!

**Comment by dvoytik** [comment]
@emilk 
I also noticed this issue...
Some more digging:
```↪ cat /proc/5245/maps | grep icu
7f9086600000-7f908835d000 r--p 00000000 103:05 4210813                   /usr/lib/libicudata.so.74.2
7f908835d000-7f908835e000 r--p 01d5c000 103:05 4210813                   /usr/lib/libicudata.so.74.2
7f9088400000-7f9088467000 r--p 00000000 103:05 4210832                   /usr/lib/libicuuc.so.74.2
7f9088467000-7f9088567000 r-xp 00067000 103:05 4210832                   /usr/lib/libicuuc.so.74.2
7f9088567000-7f90885f8000 r--p 00167000 103:05 4210832                   /usr/lib/libicuuc.so.74.2
7f90885f8000-7f908860b000 r--p 001f8000 103:05 4210832                   /usr/lib/libicuuc.so.74.2
7f908860b000-7f908860c000 rw-p 0020b000 103:05 4210832                   /usr/lib/libicuuc.so.74.2
```
That's a total of 81 MiB of private mapping. Not sure what component in egui or its dependency causing this.
Other noticeable contributors:
```
2      22M        50M                   /usr/lib/dri/iris_dri.so
3      12M        28M                   /usr/lib/libLLVM-16.so
```
The first, I suppose, is Intel user space driver. But why libLLVM requires so much private mapped memory:
```
7f9088c00000-7f908974d000 r--p 00000000 103:05 4226262                   /usr/lib/libLLVM-16.so
7f908974d000-7f908dab0000 r-xp 00b4d000 103:05 4226262                   /usr/lib/libLLVM-16.so
7f908dab0000-7f9090851000 r--p 04eb0000 103:05 4226262                   /usr/lib/libLLVM-16.so
7f9090851000-7f9090f05000 r--p 07c50000 103:05 4226262                   /usr/lib/libLLVM-16.so
7f9090f05000-7f9090f4e000 rw-p 08304000 103:05 4226262                   /usr/lib/libLLVM-16.so
```
